rust/arrow/src/array/equal/null.rs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::array::ArrayData;
+
+#[inline]
+pub fn null_equal(
+    _lhs: &ArrayData,
+    _rhs: &ArrayData,
+    _lhs_start: usize,
+    _rhs_start: usize,
+    _len: usize,
+) -> bool {
+    // a null buffer's range is always true, as every element is by definition equal (to null).
+    // We only need to compare data_types
+    true
```

Comment:
Note that this function already assumes that me performed that comparison before-hand. We do that on the `equal::equal` and `equal::equal_range` functions.

The reason I moved `base_equal` away from type-specific functions is that it allow us to have a single function per type for both range equality and equality.

To avoid the risk of being misused (i.e. not checking `base_equal` first), I pushed a commit that privatizes all these functions, effectively making `equal::equal` the only public function of the module `equal`.

Reply:
I added a test for this, just for avoidance of doubts :)